fortitude/src/rules/correctness/error_handling.rs

```diff
@@ -0,0 +1,337 @@
+            Self {
+                name,
+                stat: stat_type,
+                result: CheckStatus::Unchecked
+            },
+            &stat_node
+        ))
+    }
+
+    fn entrypoints() -> Vec<&'static str> {
+        vec![
+            "allocate_statement",
+            "deallocate_statement",
+            "open_statement",
+            "close_statement",
+            "read_statement",
+            "write_statement",
+            "inquire_statement",
+            "file_position_statement",
+            "subroutine_call", // for execute_command_line
+            "call_expression", // for wait and flush
+        ]
+    }
+}
+
+fn find_stat_in_siblings(node: &Node, stat_name: &str, src: &str) -> Result<CheckStatus> {
+    let mut sibling = *node;
+
+    while let Some(node) = sibling.next_sibling() {
+        sibling = node;
+
+        if new_branch(&sibling) {
+            // Ignore occurences in else, elseif, and case statements,
+            // as these are out of reach from the current assignment.
+            continue;
+        }
+
+        if let Some(stat_node) = once(sibling)
+            .chain(sibling.descendants())
+            .find(|d| d.kind() == "identifier" && d.to_text(src) == Some(stat_name))
+        {
+            return stat_check_status(&stat_node, stat_name, src);
+        }
+    }
+    Ok(CheckStatus::Unchecked)
+}
+
+fn stat_check_status(node: &Node, stat_name: &str, src: &str) -> Result<CheckStatus> {
+    let ancestor = node.parent().context("Node should have a parent")?;
+
+    // Two cases to consider:
+    //
+    // - The stat variable is on the left hand side of an assignment statement.
+    // - The stat variable is passed to another error handling parameter.
+    //
+    // We expect false negatives if stat is passed to a user function or
+    // subroutine and overwritten/ignored there.
+
+    if ancestor.kind() == "assignment_statement" {
+        if let Some(lhs) = ancestor.child_by_field_name("left") {
+            if lhs.to_text(src).context("to_text error")? == stat_name {
+                return Ok(CheckStatus::Overwritten);
+            }
+        }
+    }
+    if ancestor.kind() == "keyword_argument" {
+        if let Some(name_node) = ancestor.child_by_field_name("name") {
+            if is_error_checking_routine(&ancestor, src)?
+                && StatType::try_from(name_node.to_text(src).context("to_text error")?).is_ok()
+            {
+                return Ok(CheckStatus::Overwritten);
+            }
+        }
+    }
+    Ok(CheckStatus::Checked)
+}
+
+fn not_scope_boundary(node: &Node) -> bool {
+    !matches!(
+        node.kind(),
+        "function" | "subroutine" | "program" | "module_procedure" | "block_construct"
+    )
+}
+
+fn new_branch(node: &Node) -> bool {
+    matches!(
+        node.kind(),
+        "elseif_clause" | "else_clause" | "case_statement"
+    )
+}
+
+fn is_error_checking_routine(node: &Node, src: &str) -> Result<bool> {
```

Comment:
This looks very similar to `StatType::from_node` -- anyway to reuse that here?

Reply:
I've fixed it up in https://github.com/PlasmaFAIR/fortitude/pull/458/commits/1d4e4e5224e0aca4cfbe0113a8338d2d6ff4545f. Also found and fixed a lot of case-sensitivity issues while I was at it.